CMakeLists.txt

```diff
@@ -150,6 +150,12 @@ endif()
 # Kokkos recommendatation resulting in not using default GNU extensions
 set(CMAKE_CXX_EXTENSIONS OFF)
 
+# Build Kokkos as a shared library if building parthenon as a shared library (by
+# providing a custom CMAKE_INSTALL_PREFIX)
+if(NOT CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
```

Comment:
I'm not quite sure I understand why this is needed, BUILD_SHARED_LIBS should automatically propagate if parthenon is built as a shared library. And why would you want to force it so be made shared if you change the install path?

In this case if you want to build it as static and you change the install directory you are trying to build parthenon as static while Kokkos as shared? 

I'm just a little confused.

Reply:
Yes I agree with this. I was just only thinking about building a shared library, and I was getting build errors if Kokkos was built as static when I was trying to build parthenon as shared (I think this makes sense). I'll add an option to build parthenon as static/shared/none.

Reply:
Yeah, if you're building parthenon as shared, you _must_ build Kokkos as shared, too.

Yeah, don't key off of `CMAKE_INSTASLL_PREFIX_INTIAIALIZED_TO_DEFAULT`. Seems orthogonal.

Just rely on the user to set `BUILD_SHARED_LIBS`. It will then propagate to Kokkos, too. It's a standard CMake variable, not kokkos-specific: https://cmake.org/cmake/help/latest/variable/BUILD_SHARED_LIBS.html?highlight=build_shared_libs